src/qvgraphicsview.cpp

```diff
@@ -167,10 +168,20 @@ void QVGraphicsView::wheelEvent(QWheelEvent *event)
 
     if (!willZoom)
     {
-        if (event->modifiers() == (Qt::ControlModifier | Qt::ShiftModifier) || event->modifiers() == Qt::ShiftModifier)
-            translate(event->angleDelta().y()/2.0, event->angleDelta().x()/2.0);
-        else
-            translate(event->angleDelta().x()/2.0, event->angleDelta().y()/2.0);
+        const qreal scrollDivisor = 2.0; // To make scrolling less sensitive
+        qreal scrollX = event->angleDelta().x() * (horizontalScrollBar()->isRightToLeft() ? 1 : -1) / scrollDivisor;
+        qreal scrollY = event->angleDelta().y() * -1 / scrollDivisor;
+
+        if (event->modifiers() & Qt::ShiftModifier)
+            std::swap(scrollX, scrollY);
+
+        QPointF targetScrollDelta = QPointF(scrollX, scrollY) - lastScrollRoundingError;
+        QPoint roundedScrollDelta = targetScrollDelta.toPoint();
+
+        horizontalScrollBar()->setValue(horizontalScrollBar()->value() + roundedScrollDelta.x());
+        verticalScrollBar()->setValue(verticalScrollBar()->value() + roundedScrollDelta.y());
+
+        lastScrollRoundingError = roundedScrollDelta - targetScrollDelta;

```

Comment:
Scrollbars use integer values so the rounding error is tracked to avoid swallowing 1 pixel angle deltas as can happen with precision touchpads. You can see the effect of this code even with a normal mouse by setting `scrollDivisor` to 240 (2 wheel ticks), which should result in 1 pixel scrolled for every other wheel tick.